opendebates/moderator_views.py

```diff
@@ -1,60 +1,127 @@
-        to_remove.moderated = False
-    elif request.POST.get("handling") == "merge":
-        duplicate_of.keywords = (duplicate_of.keywords or '') \
-                                + " " + to_remove.idea \
-                                + " " + (to_remove.keywords or '')
-        duplicate_of.has_duplicates = True
-        duplicate_of.save()
-    if duplicate_of is not None:
-        to_remove.duplicate_of = duplicate_of
-    to_remove.save()
+        return HttpResponseNotFound()
+
+    form = ModerationForm(data=request.POST or None)
+    if request.method == 'POST':
+        if form.is_valid():
+            to_remove = form.cleaned_data['to_remove_obj']
+            duplicate_of = form.cleaned_data.get('duplicate_of_obj')
+            post_url = 'moderation_merge' if duplicate_of else 'moderation_remove'
+
+            return {
+                'form': form,
+                'to_remove': to_remove,
+                'duplicate_of': duplicate_of,
+                'post_url': post_url
+            }
+    return {'form': form}
+
 
-    if request.POST.get("handling") == "merge":
+@allow_http("POST")
+@login_required
+def merge(request):
+    if not request.user.is_superuser:
+        return HttpResponseNotFound()
+
+    to_remove = get_object_or_404(Submission, pk=request.POST['to_remove'], approved=True)
+    duplicate_of = get_object_or_404(Submission, pk=request.POST['duplicate_of'], approved=True)
+
+    if request.POST.get("action").lower() == "reject":
+        msg = _(u'No changes made and flag has been removed.')
+    elif request.POST.get("action").lower() == "merge":
         votes_already_cast = list(Vote.objects.filter(
             submission=duplicate_of).values_list("voter_id", flat=True))
         votes_to_merge = Vote.objects.filter(submission=to_remove).exclude(
             voter__in=votes_already_cast)
+        duplicate_of.keywords = (duplicate_of.keywords or '') \
+            + " " + to_remove.idea \
+            + " " + (to_remove.keywords or '')
+        duplicate_of.has_duplicates = True
         duplicate_of.votes += votes_to_merge.count()
         duplicate_of.save()
         votes_to_merge.update(original_merged_submission=to_remove, submission=duplicate_of)
-
-    if request.POST.get("send_email") == "yes":
-        if request.POST.get("handling") == "merge":
+        to_remove.duplicate_of = duplicate_of
+        to_remove.save()
+        msg = _(u'Question has been merged.')
+        if request.POST.get("send_email") == "yes":
             send_email("your_idea_is_merged", {"idea": to_remove})
-            send_email("idea_merged_into_yours", {"idea": to_remove})
-        else:
-            if duplicate_of is not None:
-                send_email("idea_is_duplicate", {"idea": to_remove})
-            else:
-                send_email("idea_is_removed", {"idea": to_remove})
-    return redirect(".")
+            send_email("idea_merged_into_yours", {"idea": duplicate_of})
```

Comment:
Can we do the latter?  ("change send_email to log the problem but not raise an exception in that particular case?")  My intention when building that feature was to fail silently if an email is missing from the database, to give the moderation team leeway to turn off specific email streams altogether (without code changes) if they decide some of the planned email triggers are not useful.